src/main/groovy/com/netflix/nebula/lint/plugin/LintRuleRegistry.groovy

```diff
@@ -57,13 +59,13 @@ class LintRuleRegistry {
             throw new InvalidRuleException(String.format("No implementation class or includes specified for rule '%s' in %s.", ruleId, ruleDescriptor))
         }
 
-        def included = includes.collect { buildRules(it as String, project, critical) }.flatten() as List<Rule>
+        def included = includes.collect { buildRules(it as String, projectSupplier, critical) }.flatten() as List<Rule>
 
         if(implClassName) {
             try {
                 Rule r = (Rule) classLoader.loadClass(implClassName).newInstance()
                 if(r instanceof ModelAwareGradleLintRule) {
-                    (r as ModelAwareGradleLintRule).project = project
+                    (r as ModelAwareGradleLintRule).projectSupplier = projectSupplier
```

Comment:
`ModelAwareGradleLintRule` rules require project access to navigate configurations/dependencies. This PR only covers basic project information, but not configurations/dependencies. This is where the approach here breaks down, since all interesting Nebula Lint rules are `ModelAwareGradleLintRule` subtypes and do require access to dependency information.